src/main/java/org/example/HttpService.java

```diff
@@ -0,0 +1,115 @@
+                    responses.add(partresponse);
+                }
+            }
+            return responses;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<FullResponse> getRandomCocktail() {
+        URI uri = URI.create(domain + BASE_PATH + "random.php");
+            JsonNode nodeArray = getNodeArrayOfResponses(uri);
+            return handleFullResponse(nodeArray);
+        }
+
+    public JsonNode getNodeArrayOfResponses(URI uri) {
+        HttpRequest request = HttpRequest.newBuilder(uri).GET().build();
+        HttpResponse<String> response;
+        JsonNode nodeArray = null;
+        try {
+            response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+            int statusCode = response.statusCode();
+            if (statusCode >= 200 && statusCode < 300) {
+                if (response.body().isBlank()){
+                    log.error("We couldn't find a cocktail that matches your request.");
+                    return null;
+                }else {
+                    log.info("Status code :" + statusCode);
+                    JsonNode node = mapper.readValue(response.body(), JsonNode.class);
+                    nodeArray = node.get("drinks");
+                    System.out.println(nodeArray);
+                }
+            }
+
+            return  nodeArray;
+        } catch (Exception e) {
+            log.error("Something went wrong");
+            throw new RuntimeException(e);
+        }
+    }
+
+    public List<FullResponse> handleFullResponse(JsonNode nodeArray) {
+        List<FullResponse> fullResponseArray = new ArrayList<>();
+        if (nodeArray!=null && nodeArray.isArray() && !nodeArray.isEmpty()) {
+            for (JsonNode objNode : nodeArray) {
+                List<String> ingredients = new ArrayList<>();
+                FullResponse fullResponse1 = new FullResponse();
+                fullResponse1.setStrDrink(objNode.get("strDrink").textValue());
+                fullResponse1.setStrCategory(objNode.get("strCategory").textValue());
+                fullResponse1.setStrAlcoholic(objNode.get("strAlcoholic").textValue());
+                fullResponse1.setStrInstructions(objNode.get("strInstructions").textValue());
+                for (int i = 0; i < 16; i++) {
+                    if (objNode.get("strIngredient" + i) != null) {
+                        if (objNode.get("strIngredient" + i).textValue() != null) {
+                            ingredients.add(objNode.get("strIngredient" + i).textValue());
+                        }
+                        i++;
+                    }
+                }
+                fullResponse1.setStrIngredient(ingredients);
+                fullResponseArray.add(fullResponse1);
+            }
+            return  fullResponseArray;
+        }
+        else {
+            log.error("No data to present");
+            return null;
+        }
+    }
```

Comment:
You are overcomplicating here, just use mapper. For this exercise we don't really need all the ingredients, so you can use first five. All others are going to be ignored. OR you can also write a custom mapper https://www.baeldung.com/jackson-deserialization (Third and fourth points)